test/test_nvfuser_frontend.py

```diff
@@ -581,5 +583,25 @@ def fusion_func(fd : FusionDefinition) :
         eager_out2 = prims.broadcast_in_dim(inputs[2], eager_out1.size(), [0, 1, 2])
         self.assertEqual(eager_out2, nvf_out)
 
+    def test_alias_output_to_input(self) :
+        def fusion_func(fd : FusionDefinition) :
+            t0 = fd.define_tensor(2)
+            s0 = fd.define_constant(1.0)
+            t1 = fd.ops.add(t0, s0)
+            t2 = fd.ops.add(t1, s0)
+            fd.add_output(t1, alias_input=t0)
+            fd.add_output(t2)
+
+        inputs = [
+            torch.ones(4, 4, device='cuda'),
+        ]
+
+        nvf_out, _ = self.exec_nvfuser(fusion_func, inputs)
```

Comment:
I agree with Ivan that there's a need to hide the aliased output in order to match the semantics. Let's put an example for that as well.

Reply:
One more thing regarding returning a reference to aliased output.

IMHO, we should try to restrict the usage of aliases. Because we don't have proper alias analysis, so it's totally not a safe thing to use and we should make it CRYSTAL CLEAR in our APIs.
Let's just NOT support returning a reference to the aliased input. So it's less likely that a general case aliased update is accidentally implemented with nvfuser python frontend.

I know it's a weak argument... We can also leave the code as-is and start a separate discussion on how to better communicate it with users.